docs/dsl_inspec.md

```diff
@@ -239,6 +239,43 @@ end
 
 Mixing this with other conditionals (like checking existence of the files etc.) can help to test different test paths using InSpec. This way you can skip certain tests which would 100% fail due to the way servers are prepared, but you know that the same test suites are reused later in different circumstances by different teams.
 
+Some notes about `only_if`:
+
+ * `only_if` applies to the entire `control`. If the results of the `only_if` block evaluate to false, the contents of the describe blocks will not be run. However, the describe block resources (`command('redis-cli SET test_inspec "HELLO"')` in the above example) that preceded the only_if WILL run. For this reason, `only_if` should usually come first in the control, prior to any `describe` blocks.
+
+```ruby
+control "don't do this" do
+  describe command("something-dangerous") do
+    # ...
+  end
+  # command("something-dangerous") already ran!!!
+  only_if { its_safe }
+end
+```
+
+Instead, do this:
+```ruby
+control "do this instead" do
+  only_if { its_safe }
+  # command("something-dangerous") won't run unless it is safe
+  describe command("something-dangerous") do
+    # ...
+  end
+end
+```
+
+ * `only_if` results in a skipped result for the control and all tests in the control, and is the only supported way of skipping an InSpec control. Other methods relying on RSpec-specific features such as `before` and `skip` may not work in the future.
+ * Only one `only_if` is permitted per `control` block; if multiple `only_if` blocks are present, only the last one will be honored.
+ * `only_if` may also be used outside a control block. In that case, it will be used to skip all controls in the current file.
```

Comment:
Is this similar as above where if you put the `only_if` in the middle of the file, it will only skip the following controls? & Only one only_if will run in the scope?

Reply:
My testing showed it skipped all controls in the file, including those that preceded it.

Reply:
That's cool, I didn't know that. Quickly testing it, I see you can also have a per-control skip as well as the entire-file skip. That's interesting because if you have one control per file then you have a way to have two separate conditions that don't share the same line of code (something we've wanted more than once in the past).